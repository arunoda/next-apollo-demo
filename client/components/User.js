const User = ({ client }) => {
  return (
    <li className="user">
      <figure>
        <p>{client.name}</p>
        <span>
          #{client.address.suite}, {client.address.street}
        </span>
        <br />
        <span>{client.address.city}</span>
        <br />
        <span>{client.address.zipcode}</span>
      </figure>
    </li>
  );
};
export default User;
