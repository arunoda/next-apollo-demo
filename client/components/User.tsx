
import React, { useEffect, useState } from 'react';
import { useQuery, gql } from '@apollo/client';
import { Card } from './Card';
import { FunctionComponent } from 'react';

const USERS = gql`
  {
      users {
      firstname
      lastname
      address
      city
      state
      country
      phone
    }
  }
`;

type ItemProps = {
  from: number, 
  qty: number
}

export const User: FunctionComponent<ItemProps> = ({ from, qty }) => {
  const { loading, error, data } = useQuery(USERS);
  const [ filteredUsers, setFilteredUsers ] = useState([])

  const getUsers =  (from: number, qty: number) => {
      const lastUser = from + qty - 1;
      const filteredUsers = data.users.filter((_item: String, index: number) => {
          return ( index <= lastUser );
      })

      return filteredUsers
  }

  useEffect(() => {
    if(data) {
      setFilteredUsers( getUsers(from, qty) );
    }
  },[data, from])

  if (loading) return <p>Loading...</p>;
  if (error) return <p>Error :(</p>;

  return (
    <div className="cards">
      <Card filteredUsers={ filteredUsers } />
    </div>
  )

}

