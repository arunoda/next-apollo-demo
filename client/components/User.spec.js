import React from "react";
import User from "./User";
import renderer from "react-test-renderer";

const data = { name: "singh", address: { suite: "88", city: "city" } };
describe("User", () => {
  it("should renders correct tree when particular props are sent", () => {
    const tree = renderer.create(<User client={data} />);
    expect(tree).toMatchSnapshot(
      `<li className="user">
        <figure>
          <p>singh</p>
          <span># 
          88 
          ,</span>
          <br />
          <span>city</span>
          <br />
          <span />
        </figure>
      </li>`
    );
  });
});
