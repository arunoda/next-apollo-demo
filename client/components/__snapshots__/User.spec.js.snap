// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User renders correct tree when particular props are sent: <li className="user">
        <figure role="usercard">
          <p>singh</p>
          <span># 
          88 
          ,</span>
          <br />
          <span>city</span>
          <br />
          <span />
        </figure>
      </li> 1`] = `
<li
  className="user"
>
  <figure
    role="usercard"
  >
    <p>
      singh
    </p>
    <span>
      #
      88
      , 
    </span>
    <br />
    <span>
      city
    </span>
    <br />
    <span />
  </figure>
</li>
`;

exports[`User should renders correct tree when particular props are sent: <li className="user">
        <figure role="usercard">
          <p>singh</p>
          <span># 
          88 
          ,</span>
          <br />
          <span>city</span>
          <br />
          <span />
        </figure>
      </li> 1`] = `
<li
  className="user"
>
  <figure
    role="usercard"
  >
    <p>
      singh
    </p>
    <span>
      #
      88
      , 
    </span>
    <br />
    <span>
      city
    </span>
    <br />
    <span />
  </figure>
</li>
`;

exports[`User should renders correct tree when particular props are sent: <li className="user">
        <figure>
          <p>singh</p>
          <span># 
          88 
          ,</span>
          <br />
          <span>city</span>
          <br />
          <span />
        </figure>
      </li> 1`] = `
<li
  className="user"
>
  <figure>
    <p>
      singh
    </p>
    <span>
      #
      88
      , 
    </span>
    <br />
    <span>
      city
    </span>
    <br />
    <span />
  </figure>
</li>
`;
